spec/models/dataset_spec.rb

```diff
@@ -103,4 +103,29 @@
     expect(dataset.last_published_at).to eq second_publish
   end
 
+  it "only updates legacy with datafiles that have been updated" do
```

Comment:
maybe something more detailed, like "only updates legacy with datafiles that have been updated since latest dataset publication" or something along these lines.